src/CloudEventFunctionWrapper.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use RuntimeException;
+
+class CloudEventFunctionWrapper extends FunctionWrapper
+{
+    public function execute(ServerRequestInterface $request): ResponseInterface
+    {
+        // Get Body
+        $body = (string) $request->getBody();
+        $cloudeventData = json_decode($body, true);
+        if (json_last_error() != JSON_ERROR_NONE) {
+            throw new RuntimeException(sprintf(
+                'Could not parse CloudEvent: %s',
+                '' !== $body ? json_last_error_msg() : 'Missing cloudevent payload'
```

Comment:
`'' != $body` reads unusual, but looking again I see we've already cast this value to a string so the edge case concerns to use `empty()` don't apply.

Nicely concise.

Alternate consideration: PHP 7.3 introduced `JSON_THROW_ON_ERROR` which may be tidier.

Reply:
We have this library as a minimum of 7.1. We could (and probably should) raise this to 7.2, but as GAE runs php72, I would not want to raise minimum requirements past that.

Reply:
This is the same as in our background functions wrapper.

Going to keep as is unless we want to change in both places now.